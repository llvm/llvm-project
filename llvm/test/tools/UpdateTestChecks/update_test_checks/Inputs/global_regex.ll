; RUN: opt < %s -S | FileCheck %s

@Bar = global i32 0
@OnlyFoo = global i32 1
@Baz = global i32 2

define i32 @t() {
  %v1 = load i32, ptr @Bar
  %v2 = load i32, ptr @OnlyFoo
  %v3 = load i32, ptr @Baz
  %a1 = add i32 %v1, %v2
  %a2 = add i32 %a1, %v3
  ret i32 %a2
}

