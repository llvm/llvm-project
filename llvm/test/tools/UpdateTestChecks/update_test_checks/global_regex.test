# REQUIRES: x86-registered-target
## Basic test checking that update_test_checks.py works correctly
# RUN: cp -f %S/Inputs/global_regex.ll %t.ll && %update_test_checks %t.ll --check-globals --global-value-regex "OnlyFoo"
# RUN: diff -u %t.ll %S/Inputs/global_regex.ll.expected
## Verify that running without the --global-value-regex flag respects UTC_ARGS
# RUN: %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/global_regex.ll.expected
