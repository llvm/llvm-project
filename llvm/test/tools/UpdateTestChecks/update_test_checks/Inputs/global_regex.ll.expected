; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --check-globals --global-value-regex "OnlyFoo"
; RUN: opt < %s -S | FileCheck %s

@Bar = global i32 0
@OnlyFoo = global i32 1
@Baz = global i32 2

;.
; CHECK: @[[ONLYFOO:[a-zA-Z0-9_$"\\.-]+]] = global i32 1
;.
define i32 @t() {
; CHECK-LABEL: @t(
; CHECK-NEXT:    [[V1:%.*]] = load i32, ptr @Bar, align 4
; CHECK-NEXT:    [[V2:%.*]] = load i32, ptr @OnlyFoo, align 4
; CHECK-NEXT:    [[V3:%.*]] = load i32, ptr @Baz, align 4
; CHECK-NEXT:    [[A1:%.*]] = add i32 [[V1]], [[V2]]
; CHECK-NEXT:    [[A2:%.*]] = add i32 [[A1]], [[V3]]
; CHECK-NEXT:    ret i32 [[A2]]
;
  %v1 = load i32, ptr @Bar
  %v2 = load i32, ptr @OnlyFoo
  %v3 = load i32, ptr @Baz
  %a1 = add i32 %v1, %v2
  %a2 = add i32 %a1, %v3
  ret i32 %a2
}

